polkadot/xcm/src/v4/mod.rs

```diff
@@ -1016,6 +1016,9 @@ pub enum Instruction<Call> {
 	///
 	/// Errors: If the given origin is `Some` and not equal to the current Origin register.
 	UnpaidExecution { weight_limit: WeightLimit, check_origin: Option<Location> },
+
+	/// A directive to indicate some arbitrary key value pair wants to be published somewhere.
+	Publish { data: Vec<Vec<u8>, Vec<u8>> },
```

Comment:
I dont know how XCM normally handles this or if it makes sense, but maybe we can use a `BoundedVec` to avoid the case that the receiver chain cannot decode it?

Reply:
I opened an RFC for this to discuss all these things: https://github.com/paritytech/xcm-format/pull/48
It's probably safer for it to be a boundedvec